tests/foreman/cli/test_contentviewfilter.py

```diff
@@ -15,131 +15,132 @@
 :Upstream: No
 """
 import pytest
-from fauxfactory import gen_choice
 from fauxfactory import gen_string
 
 from robottelo.cli.base import CLIReturnCodeError
 from robottelo.cli.contentview import ContentView
 from robottelo.cli.defaults import Defaults
 from robottelo.cli.factory import make_content_view
-from robottelo.cli.factory import make_content_view_filter
-from robottelo.cli.factory import make_location
-from robottelo.cli.factory import make_org
-from robottelo.cli.factory import make_product_wait
 from robottelo.cli.factory import make_repository
 from robottelo.cli.repository import Repository
 from robottelo.constants import DOCKER_REGISTRY_HUB
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
-from robottelo.test import CLITestCase
 
 
-class ContentViewFilterTestCase(CLITestCase):
-    """Content View Filter CLI tests"""
+@pytest.fixture(scope='module')
+def repo(module_org, module_product):
+    return make_repository({'organization-id': module_org.id, 'product-id': module_product.id})
 
-    @classmethod
-    def setUpClass(cls):
-        """Init single organization, product and repository for all tests"""
-        super().setUpClass()
-        cls.org = make_org()
-        cls.product = make_product_wait({'organization-id': cls.org['id']})
-        cls.content_view = make_content_view({'organization-id': cls.org['id']})
-        for _ in range(2):
-            cls.repo = make_repository({'product-id': cls.product['id']})
-            Repository.synchronize({'id': cls.repo['id']})
-            ContentView.add_repository(
-                {'id': cls.content_view['id'], 'repository-id': cls.repo['id']}
-            )
+
+@pytest.fixture
+def content_view(module_org, repo):
+    content_view = make_content_view({'organization-id': module_org.id})
+
+    Repository.synchronize({'id': repo['id']})
```

Comment:
This is now syncing the repository at a function scope, instead of a class scope, which could significantly impact run time. I believe you can class or even module scope this fixture.

Reply:
Moved the repo sync into the repo fixture, and renamed it `sync_repo`. This actually makes it consistent with the corresponding code in `tests/foreman/api/test_contentviewfilter.py`.